src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,166 @@
+
+      m_pidControllerArm = m_Arm.getPIDController();
+      m_pidControllerArm.setFeedbackDevice(m_AbsEncoder);
+
+      
+    NetworkTable ArmTuningTable = NetworkTableInstance.getDefault().getTable(m_tuningTable);
+    m_ArmPSubs = ArmTuningTable.getDoubleTopic("P").getEntry(Config.ArmConfig.arm_kP, PubSubOption.periodic(0.02));
+    m_ArmISubs = ArmTuningTable.getDoubleTopic("I").getEntry(Config.ArmConfig.arm_kI);
+    m_ArmDSubs = ArmTuningTable.getDoubleTopic("D").getEntry(Config.ArmConfig.arm_kD);
+    m_ArmIzSubs = ArmTuningTable.getDoubleTopic("IZone").getEntry(Config.ArmConfig.arm_kIz);
+    m_ArmFFSubs = ArmTuningTable.getDoubleTopic("FF").getEntry(Config.ArmConfig.arm_kFF);
+   //m_topArmOffset = topArmTuningTable.getDoubleTopic("Offset").getEntry(ArmConfig.top_arm_offset);
+    momentToVoltageConversion = ArmTuningTable.getDoubleTopic("VoltageConversion").subscribe(m_VoltageConversion);
+
+    m_ArmFFSubs.setDefault(Config.ArmConfig.arm_kFF);
+    m_ArmPSubs.setDefault(Config.ArmConfig.arm_kP);
+    m_ArmISubs.setDefault(Config.ArmConfig.arm_kI);
+    m_ArmDSubs.setDefault(Config.ArmConfig.arm_kD);
+    m_ArmIzSubs.setDefault(Config.ArmConfig.arm_kIz);
+
+    NetworkTable ArmDataTable = NetworkTableInstance.getDefault().getTable(m_dataTable);
+
+    
+    m_ArmPosPub = ArmDataTable.getDoubleTopic("MeasuredAngleDeg").publish(PubSubOption.periodic(0.02));
+    m_TargetAngle = ArmDataTable.getDoubleTopic("TargetAngle").publish(PubSubOption.periodic(0.02));
+
+    ErrorCheck.errREV(m_pidControllerArm.setP(m_ArmPSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setI(m_ArmISubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setD(m_ArmDSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setIZone(m_ArmIzSubs.get())); 
+    ErrorCheck.errREV(m_pidControllerArm.setOutputRange(Config.ArmConfig.min_output, Config.ArmConfig.max_output));
+
+    updatePIDSettings();
+    //updateFromAbsoluteBottom();
+  }
+  public void updatePIDSettings() {
+    ErrorCheck.errREV(m_pidControllerArm.setFF(m_ArmFFSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setP(m_ArmPSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setI(m_ArmISubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setD(m_ArmDSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setIZone(m_ArmIzSubs.get()));
+    ErrorCheck.errREV(m_pidControllerArm.setOutputRange(Config.ArmConfig.min_output, Config.ArmConfig.max_output));
+  }
+  @Override
+  public void periodic() {
+    m_ArmPosPub.accept(Math.toDegrees(m_AbsEncoder.getPosition()));
+    m_ArmVelPub.accept(m_AbsEncoder.getVelocity());
+  }
+    //input angle_bottom in radians
+    public void setJointAngle(double angle) {
+      if (angle<Math.toRadians(Config.ArmConfig.MIN_ARM_ANGLE_DEG) || angle>Math.toRadians(Config.ArmConfig.MAX_ARM_ANGLE_DEG)) {
+        angle = Math.toRadians(Config.ArmConfig.MAX_ARM_ANGLE_DEG);
+      }
```

Comment:
This if statement can be replaced by the clamp function from WPILib. `MathUtil.clamp(...)`